### PR TITLE
[TensorRT EP] Customizable engine cache prefix

### PR DESCRIPTION
**Pull request by yf711**
### Description

Add new option `trt_engine_cache_prefix` to customize TRTEP engine cache prefix.

i.e:
- If user specifies `trt_engine_cache_prefix|FRCNN trt_engine_cache_enable|true` when running FRCNN model 
  - the cache will be saved/loaded: `FRCNN_2068723788287043730_*_sm80.engine`. Engine profile follows same pattern.

- If skipping this option, the engine will be saved/loaded: `TensorrtExecutionProvider_TRTKernel_graph_torch-jit-export_2068723788287043730_*_*_sm80.engine` as default case.

### Motivation and Context

https://github.com/microsoft/onnxruntime/issues/16708

### CODE REVIEW

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h [hunk 1]
```diff
@@ -288,8 +291,9 @@ class TensorrtExecutionProvider : public IExecutionProvider {
   int (*engine_encryption_)(const char*, char*, size_t) = nullptr;
   bool timing_cache_enable_ = false;
   bool force_timing_cache_match_ = false;
-  bool detailed_build_log_ = false;
+  bool detailed_build_log_ = true;
```
Comment (chilo-ms) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h:294]: why changing it to true?

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc [hunk 2]
```diff
@@ -2769,12 +2772,14 @@ Status TensorrtExecutionProvider::CreateNodeComputeInfoFromGraph(const GraphView
     }
   }
 
+  // Generate cache suffix in case user would like to customize cache prefix
+  std::string cache_suffix = "_" + GetCacheSuffix(fused_node.Name(), trt_node_name_with_precision);
```
Comment (chilo-ms) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:2776]: It's better to compose the cache_suffix only if user specifies cache_prefix.
You can move this line down to where it check "!cache_prefix_.empty()"

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h (hunk 1 above)
Comment (yf711) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h]: That was for testing purpose and I have changed it back. Thanks for pointing that out

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc (hunk 2 above)
Comment (yf711) [reply onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc]: Thanks! This has been applied

File: onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_utils.h [hunk 3]
```diff
@@ -695,4 +697,49 @@ bool ParseProfileShapes(std::string profile_shapes_string, std::unordered_map<st
...
+    }
+  }
+  return result;
+}
+
+/*
+ * Parse engine cache name suffix when user customizes prefix for engine cache name
+ *
+ * For example:
+ * When default subgraph name is "TensorrtExecutionProvider_TRTKernel_graph_torch-jit-export_2068723788287043730_189_189_fp16"
```
Comment (chilo-ms) [inline onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_utils.h:726]: Minor suggestion. No need to do here.
As you can see the subgraph name has duplicate information of subgraph index string "189".
One of the subgraph index string is added by [TRT EP](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L1971) and other subgraph index string i'm guessing is added by ORT core.
If somehow we can manage to remove one of the duplicate subgraph index string, the GetCacheSuffix() might be easier to parse the string.

**Review by chilo-ms — APPROVED** [review]

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**